library/src/main/java/kwasm/format/text/token/IntegerLiteral.kt

```diff
@@ -54,8 +54,8 @@ sealed class IntegerLiteral<Type>(
 ) : Token {
     val value: Type by lazy {
         val res = parseValue()
-        if (!checkMagnitude(res, magnitude)) {
-            throw ParseException("Illegal value: $res, for expected magnitude: $magnitude", context)
+        if (!checkMagnitude(res, this.magnitude)) {
+            throw ParseException("Illegal value: $res, for expected magnitude: ${this.magnitude}", context)
```

Comment:
This is the bit that fixed the magnitude issue. What was going on? I don't know.

Reply:
Some nonsense with closures and the default constructor, i'm sure.